ios/brave-ios/Sources/AIChat/Components/Markdown/Code Highlight/HighlightJS.swift

```diff
@@ -11,40 +12,45 @@ import SwiftUI
 import os.log
 
 class HighlightJS {
-  private let context: JSContext
-  private let theme: [BasicCSSParser.CSSStyle]
-  private let script: String
+  private var isDarkTheme: Bool
+  private(set) var isHighlighterPrepared: Bool = false
+  private var context: JSContext?
+  private var theme: [BasicCSSParser.CSSStyle]?
+  private var script: String?
 
   static let dark = HighlightJS(darkTheme: true)
   static let light = HighlightJS(darkTheme: false)
 
   private init(darkTheme: Bool) {
-    let styleSheet =
-      HighlightJS.loadScript(named: "atom-one-\(darkTheme ? "dark" : "light").min", type: "css")
-      ?? ""
-    theme = BasicCSSParser.parse(styleSheet)
-    script = HighlightJS.loadScript(named: "highlight.min", type: "js") ?? ""
-
-    context = JSContext()
-    context.evaluateScript("var window = {};\n\n\(script)")
+    isDarkTheme = darkTheme
   }
 
-  private static func loadScript(named: String, type: String) -> String? {
-    guard let path = Bundle.module.path(forResource: named, ofType: type),
-      let source = try? String(contentsOfFile: path)
+  func prepareHighlighter() async {
+    if context != nil {
+      return
+    }
+    let scriptName = "atom-one-\(isDarkTheme ? "dark" : "light").min"
+    guard let stylesheetPath = Bundle.module.url(forResource: scriptName, withExtension: "css"),
+      let scriptPath = Bundle.module.url(forResource: "highlight", withExtension: "js"),
+      let stylesheet = await AsyncFileManager.default.utf8Contents(at: stylesheetPath),
+      let script = await AsyncFileManager.default.utf8Contents(at: scriptPath)
     else {
-      Logger.module.error("Failed to load script: \(named).js")
-      assertionFailure("Failed to Load Script: \(named).js")
-      return nil
+      return
     }
-    return source
+    theme = BasicCSSParser.parse(stylesheet)
+    context = JSContext()
+    context?.evaluateScript("var window = {};\n\n\(script)")
```

Comment:
Once this is all done, should `isHighlighterPrepared` be set to true?

Reply:
This was unused, removed in https://github.com/brave/brave-core/pull/24555/commits/0fcf83b71adffb28e3db477915d2658af50c966b